### ISSUE TITLE
[BUG]: Disney's UP TLB Miss

### ISSUE THREAD

**Issue by seta-san** [PCSX2/pcsx2#4760]
### Describe the Bug

TLB Miss on Disney's Up when entering a dark cave. 

### Reproduction Steps

3rd level. 

### Expected Behavior

emulator doesn't crash

### PCSX2 Revision

dev-1754

### Operating System

Windows 10 (64bit)

### CPU

i5-9600

### GPU

gtx 960

### GS Settings

softwar renderer

### GS Window Screenshots

![tlb miss](https://user-images.githubusercontent.com/8851653/132960297-3b2f4b2c-ed08-42a4-ad9f-486270b6a8c7.jpg)

### Logs & Dumps

C:\Users\\source\repos\PCSX2\pcsx2\pcsx2\FiFo.cpp(48) : assertion failed:
    Function:  ReadFIFO_VIF1
    Thread:    EE Core
    Condition: vif1Regs.stat.FQC != 0
    Message:   FQC = 0 on VIF FIFO READ!

Stacktrace:
[00] _hwRead128<5>                                \source\repos\PCSX2\pcsx2\pcsx2\HwRead.cpp:392
[01] _hwRead32<5,0>                               \source\repos\PCSX2\pcsx2\pcsx2\HwRead.cpp:76
[02] _hwWrite16<5>                                \source\repos\PCSX2\pcsx2\pcsx2\HwWrite.cpp:363
[03] m_IndirectDispatchers                       
[04] AppCoreThread::DoCpuExecute                  \source\repos\PCSX2\pcsx2\pcsx2\gui\AppCoreThread.cpp:677
[05] SysCoreThread::ExecuteTaskInThread           \source\repos\PCSX2\pcsx2\pcsx2\System\SysCoreThread.cpp:300
[06] Threading::pxThread::_try_virtual_invoke     \source\repos\PCSX2\pcsx2\common\ThreadTools.cpp:616
[07] Threading::pxThread::_internal_execute       \source\repos\PCSX2\pcsx2\common\ThreadTools.cpp:671
[08] Threading::pxThread::internal_callback_helper \source\repos\PCSX2\pcsx2\common\ThreadTools.cpp:722
[09] Threading::pxThread::_internal_callback      \source\repos\PCSX2\pcsx2\common\ThreadTools.cpp:711
[10] ptw32_threadStart                            \source\repos\PCSX2\pcsx2\3rdparty\pthreads4w\ptw32_threadStart.c:184
[11] o_iswdigit                                  
[12] BaseThreadInitThunk                         
[13] RtlGetAppContainerNamedObjectPath           
[14] RtlGetAppContainerNamedObjectPath

**Comment by refractionpcsx2** [comment]
Can you try the EE Timing Hack please?

**Comment by prafullpcsx2** [comment]
I have the game. A savestate right before the crash will help figure this out.

**Comment by seta-san** [comment]
[SLUS-21864 (F78D6593).00.zip](https://github.com/PCSX2/pcsx2/files/7148479/SLUS-21864.F78D6593.00.zip)
the way i've been triggering it is that i just into the cave with the old man. i get attacked by some bats. i just back out onto the little island. switch to the boy and use the mirror light and start shining it into the cave and it eventually crashes.

**Comment by seta-san** [comment]
to be fair it seems like it does it anytime the boy the uses the mirror long enough.

**Comment by gilderoylockhart01** [comment]
The problem with TLB misses is they can be cause by virtually anything.

I sugest youbfiest try clamping/rounding, then disable intant VU1 or play with the cyclerate to see if that makes a difference.

Lastly, try the EE interpreter with the EE cacje even if, well, I doubt it's this.

**Comment by prafullpcsx2** [comment]
I wonder if you uploaded the right savestate? I can use the mirror no problem but to enter the cave you need a lantern.

**Comment by seta-san** [comment]
yes. bring the mirror out and shine it for a long while. i've disabled the frame limiting.

**Comment by prafullpcsx2** [comment]
Ok, I can reproduce the tlb miss now.

**Comment by prafullpcsx2** [comment]
@seta-san 
[f78d6593.zip](https://github.com/PCSX2/pcsx2/files/7149131/f78d6593.zip)

Can you test this patch, see if it fixes the crash and doesn't cause any side effects?

**Comment by gilderoylockhart01** [comment]
What are you trying to patch in the game code prafull?

**Comment by seta-san** [comment]
first one locks up and does this and does this in an infinite loop but doesn't actually tlb miss
"[IOP] looping mode
padman: DMA Busy ID = 3c850008 ret = 0
        SB_STAT = f0000102

second one tlb misses
"

**Comment by gilderoylockhart01** [comment]
seta-san, please do a blockdump of your game as well as a savestate before the problem happens.

**Comment by refractionpcsx2** [comment]
Okay so I had a quick look at this.

I can stop it crashing by removing the forced "nearest" rounding on the FPU for DIV, then set my EE round mode to "Negative" then the crashes stop, but you do get a single frame which looks like a zebra skin (possibly some sort of depth download).  So it's not perfect, but it seems to stop the crashing.

**Comment by gilderoylockhart01** [comment]
That makes sense regarding the engine, it's also worth mentioning any asobo game will require positive VU rounding to work properly.

**Comment by refractionpcsx2** [comment]
Well, positive rounding does nothing, in this case. It still crashes.

**Comment by seta-san** [comment]
i set vu rounding to positive and ee round mode to negative and it still tlb misses

**Comment by seta-san** [comment]
and i've played up to this point in the game without issue without setting vu to positive

**Comment by gilderoylockhart01** [comment]
Usually asobo games will produce some very slight sps which can be insanelly hard to see with normal VU rounding.

As for the crashing, I don't have a dev build around but are you sure cop2 timings are right refraction? Because it's possible negative rounding hackfix the glitch like it does on ffx

**Comment by refractionpcsx2** [comment]
Nothing comes up in the log to say there's a problem with COP2 timings.

**Comment by gilderoylockhart01** [comment]
Ok then. Interesting still the game have issues like this, I hope it doesn't require softfloat.

**Comment by LoStraniero91** [comment]
Just asking, have you unlocked the lantern before entering the cave? I'm unable to reproduce the bug in the PAL version, probably because I have the lantern...

**Comment by refractionpcsx2** [comment]
Is this reproducable at all anymore? I tried at the beginning of the game where I got it to happen last time, and it's still not doing it.

if somebody can reproduce it, then I will need a savestate some time before it happens. Otherwise I'll go ahead and close this.

**Comment by seta-san** [comment]
I’ll check it when my son allows

**Comment by seta-san** [comment]
[s.zip](https://github.com/PCSX2/pcsx2/files/11452465/s.zip)
![Untitled](https://github.com/PCSX2/pcsx2/assets/8851653/748a3f24-25b7-41d1-8de6-5086c1758706)

**Comment by seta-san** [comment]
it can help to unlock frames to speed it up

**Comment by refractionpcsx2** [comment]
What was the original savestate name? You renamed it.

**Comment by seta-san** [comment]
i just saved it to file

**Comment by seta-san** [comment]
slus-21864

**Comment by refractionpcsx2** [comment]
Thanks

**Comment by refractionpcsx2** [comment]
That savestate is not crashing for me :/ I'm moving my mirror around all over the place.

Am I missing something? :/

**Comment by refractionpcsx2** [comment]
Can you try it on my True Crime PR at all? I'm trying it myself but I can't get it to trigger on there, but not sure if I'm just being fortunate.

**Comment by refractionpcsx2** [comment]
bummer, alright thanks.

**Comment by Buzzardsoul** [comment]
I reproduced it on latest. Still an issue.

**Comment by ScottNash042** [comment]
still an issue using v2.7.42
